nautobot_chatops/dispatchers/base.py

```diff
@@ -37,6 +38,63 @@ def __init__(self, context=None):
         """Init this Dispatcher with the provided dict of contextual information (which will vary by app)."""
         self.context = context or {}
 
+    def _get_cache_key(self) -> str:
+        """Key generator for the cache, adding the plugin prefix name."""
+        # Using __file__ as a key customization within the cache
+        return "-".join([__file__, self.context.get("user_id", "generic")])
+
+    def get_session_entry(self, key: str):
+        """Return the session data for a key."""
+        session_value = cache.get(self._get_cache_key()) or {}
+        try:
+            return session_value[key]
+        except KeyError:
+            return None
```

Comment:
Looks like the `cache` object works basically the same as a dict, so this could be a one-liner:

```suggestion
        return cache.get(self._get_cache_key(), {}).get(key, None)
```